app/models/node.rb

```diff
@@ -1070,6 +1070,8 @@ def can_tag(tagname, user, errors = false)
       errors ? I18n.t('node.page_does_not_exist') : false
     elsif socials[one_split&.to_sym].present?
       errors ? "This tag is used for associating a #{socials[one_split.to_sym]} account. <a href='https://publiclab.org/wiki/oauth'>Click here to read more </a>" : false
+    elsif user.first_time_poster && !(user.username == self.author.username || (self.coauthors && self.coauthors.exists?(username: user.username)) || user.role == 'admin' || user.role == 'moderator')
```

Comment:
This expression can be changed to 
```ruby
elsif user.first_time_poster && !(user.username == self.author.username || self.coauthors&.exists?(username: user.username) || logged_in_as(['admin', 'moderator']))
```
which looks a bit more simpler.

Reply:
I can't write the test myself, but I think it would be a good idea to have a test around this if one doesn't already exist :)

Reply:
> I can't write the test myself, but I think it would be a good idea to have a test around this if one doesn't already exist :)

Thanks, @lonwabo-mnyaiza I'm already writing the tests. Failing functional tests in this PR were intentional and you don't need to worry but after making this change, I'm not sure why `logged_in_as(['admin', 'moderator'])` doesn't work here. Maybe it's because `can_tag` is a property in the models file. 

Would you mind changing the last condition to `(['admin', 'moderator'].include? user.role)` instead of `logged_in_as(['admin', 'moderator'])`?
The expression would be 
```ruby
    elsif user.first_time_poster && !(user.username == self.author.username || self.coauthors&.exists?(username: user.username) || (['admin', 'moderator'].include? user.role))
```

Apologies for the last suggested change. Let me know if you need any help.
@RuthNjeri would you like to comment on this?

Reply:
@17sushmita regarding the function call to `logged_in_as`, I'm happy to make the change at 17:00 GTM. 
I'm also not 100% sure why it wouldn't work, as it is been used in similar ways on other components ¯\_(ツ)_/¯

Reply:
Hi @17sushmita and @lonwabo-mnyaiza I think the reason `logged_in_as` is not working is because it is a function in the applications controller and we are trying to use it in a model file... `node.rb`...if you want to use it in the node.rb file you need to include the application controller in it which would be a wrong way to connect a controller and model...


The alternative you have provided @17sushmita, `(['admin', 'moderator'].include? user.role)` looks great!